src/applications/hca/components/FormFields/VaMedicalCenter.jsx

```diff
@@ -81,19 +150,16 @@ const VaMedicalCenter = props => {
         isLoading(false);
       }
     },
-    [facilityState],
+    [localDataFacilityState, previousFacilityState],
```

Comment:
Not sure why it was complaining about `localData['view:facilityState']` being a "complex expression" - it wasn't qualitatively different from `props.source` in https://legacy.reactjs.org/docs/hooks-reference.html#conditionally-firing-an-effect, right? Maybe it's the `object['attributename']` instead of `object.attributename` that's tripping it up? Seems happy with it extracted out to a local `const`.

Reply:
Contrary to the Warning, we don't want `localData` in the deps here, though - otherwise, changing the _facility_ causes the component to rerender.

Reply:
Ehh CodeClimate is very opinionated so we run into these kinds of things a lot.